h5netcdf/core.py

```diff
@@ -17,13 +17,20 @@ def _reverse_dict(dict_):
 
 class BaseVariable(object):
 
-    def __init__(self, parent, h5ds, dimensions=None):
+    def __init__(self, parent, name, dimensions=None):
         self._parent = parent
-        self._h5ds = h5ds
+        self._root = parent._root
+        self._h5path = '/'.join([parent.name.rstrip('/'), name.lstrip('/')])
```

Comment:
The `h5py.Dataset` object is no longer passed to the variable class. It is recovered on-demand from the `h5py.File` object on line 28.